app/controllers/CyaMaintainProfileController.scala

```diff
@@ -104,6 +103,41 @@ class CyaMaintainProfileController @Inject() (
       case Left(errors) =>
         Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
     }
+  }
 
+  def onPageLoadNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
```

Comment:
None of the changes in this file are tested!

Reply:
I shall take a look in a moment :)